luarules/gadgets/unit_missile_counter.lua

```diff
@@ -0,0 +1,72 @@
+local gadget = gadget ---@type Gadget
+
+function gadget:GetInfo()
+	return {
+		name = "Missile counter",
+		desc = "Checks how many missiles are needed to kill a target (used on Legion Medusa)",
+		author = "EnderRobo",
+		date = "May 7, 2025",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
+
+local isCommander = {}
+
+local function missile_counter(unitID, unitDefID, team)
+	
+	local armored, armorMultiplier
+	local damage = Spring.GetUnitWeaponDamages (unitID, 2, Game.armorTypes.standard)
+	local health, maxHealth
+	local counter
+	local losstate
+	local works, isUserTarget, targetID = Spring.GetUnitWeaponTarget (unitID, 1) 
+	local burstrate = (Spring.GetUnitWeaponState (unitID, 2, "burstRate") * 1000)
+	local targetDefID
+	if type(targetID) == "number" then
+		losstate = Spring.GetUnitLosState(targetID, team, true)
+		targetDefID = Spring.GetUnitDefID(targetID)
+		
+		if isCommander[targetDefID] then																					--if its a commander fire a full volley
+			counter = 6
+			
+		elseif losstate == 6 and not UnitDefs[targetDefID].isBuilding then													--fire full volley at radar blobs
+			counter = 6
+			
+		elseif losstate == 2 then																							--fire full volley at radar blobs
+			counter = 6
+			
+		elseif losstate == (15 or 1) then																					--check how many missiles for visible target
+			armored, armorMultiplier = Spring.GetUnitArmored(targetID)
+			health = Spring.GetUnitHealth(targetID)
+			if armored then
+				counter = math.floor(health / (damage * armorMultiplier)) + 1												--account for armored state
+			else
+				counter = math.floor(health / damage) + 1	
+			end	
+			
+		elseif losstate == (14) or (losstate == 6 and UnitDefs[targetDefID].isBuilding) then								--check how many missiles for radar ghost target at max health
+			maxHealth = UnitDefs[targetDefID].health
+			counter = math.floor(maxHealth  / damage) + 1
+			
+		end
+	else																													--fire full volley at ground
+		counter = 6
+	end
+	Spring.CallCOBScript(unitID, "missile_counter", 7, 1, counter, targetID, burstrate, health, losstate, damage, armored)
+	return 1
+end
+
+function gadget:Initialize()
+	gadgetHandler:RegisterGlobal("missile_counter", missile_counter)
```

Comment:
```suggestion
	gadgetHandler:RegisterGlobal("MissileCounter", missileCounter)
```
Convention is globals are PascalCase